powerline/segments/common/players.py

```diff
@@ -477,3 +477,70 @@ def get_player_status(self, pl):
 
 {0}
 ''').format(_common_args.format('rdio')))
+
+
+
+class iTunesPlayerSegment(PlayerSegment):
+        def get_player_status(self, pl):
+                status_delimiter = '-~`/='
+                ascript = '''
+                tell application "System Events"
+                set process_list to (name of every process)
+                end tell
+                
+                if process_list contains "iTunes" then
+                tell application "iTunes"
+                if player state is playing then
+                set t_title to name of current track
+                set t_artist to artist of current track
+                set t_album to album of current track
+                set t_duration to duration of current track
+                set t_elapsed to player position
+                set t_state to player state
+                return t_title & "{0}" & t_artist & "{0}" & t_album & "{0}" & t_elapsed & "{0}" & t_duration & "{0}" & t_state
+                end if
+                end tell
+                end if
+                '''.format(status_delimiter)
+                now_playing = asrun(pl, ascript)
+                if not now_playing:
+                        return
+                now_playing = now_playing.split(status_delimiter)
+                if len(now_playing) != 6:
+                        return
+                title, artist, album = now_playing[0], now_playing[1], now_playing[2]
+                #title = title[0:15] + "…" if len(title) > 15 else title
+                #artist = artist[0:10] + "…" if len(artist) > 10 else artist
+                #album = album[0:10] + "…" if len(album) > 10 else album
+                #Modified by Rui Hopefully it works
+                if len(title) > 15:
+                        title = title[0:15] + "..."
+                else:
+                        title
+                if len(artist) > 10:
+                        artist = artist[0:10] + "..."
+                else:
+                        artist
+                if len(album) > 10:
+                        album = album[0:10] + "..."
+                else:
+                        album                        
```

Comment:
First: all `else` clauses are completely unneeded, you do not do *anything* at all here. Second: if you need truncation, truncate in format string. Truncations from here must be removed completely, also with preceding comment.